openpype/hosts/houdini/plugins/publish/collect_vray_rop.py

```diff
@@ -0,0 +1,161 @@
+
+        default_prefix = evalParmNoFrame(rop, "SettingsOutput_img_file_path")
+        render_products = []
+        # TODO: add render elements if render element
+
+        beauty_product = self.get_beauty_render_product(default_prefix)
+        render_products.append(beauty_product)
+        files_by_aov = {
+            "RGB Color": self.generate_expected_files(instance,
+                                                      beauty_product)}
+
+        if instance.data.get("RenderElement", True):
+            render_element = self.get_render_element_name(rop, default_prefix)
+            if render_element:
+                for aov, renderpass in render_element.items():
+                    render_products.append(renderpass)
+                    files_by_aov[aov] = self.generate_expected_files(instance, renderpass)          # noqa
+
+        for product in render_products:
+            self.log.debug("Found render product: %s" % product)
+            filenames = list(render_products)
+            instance.data["files"] = filenames
+        self.log.debug("files:{}".format(render_products))
+
+        # For now by default do NOT try to publish the rendered output
+        instance.data["publishJobState"] = "Suspended"
+        instance.data["attachTo"] = []      # stub required data
+
+        if "expectedFiles" not in instance.data:
+            instance.data["expectedFiles"] = list()
+        instance.data["expectedFiles"].append(files_by_aov)
+        self.log.debug("expectedFiles:{}".format(files_by_aov))
+
+    def get_beauty_render_product(self, prefix, suffix="<reName>"):
+        """Return the beauty output filename if render element enabled
+        """
+        aov_parm = ".{}".format(suffix)
+        beauty_product = None
+        if aov_parm in prefix:
+            beauty_product = prefix.replace(aov_parm, "")
+        else:
+            beauty_product = prefix
+
+        return beauty_product
+
+    def get_render_element_name(self, node, prefix, suffix="<reName>"):
+        """Return the output filename using the AOV prefix and suffix
+        """
+        render_element_dict = {}
+        # need a rewrite
+        re_path = node.evalParm("render_network_render_channels")
+        if re_path:
+            node_children = hou.node(re_path).children()
+            for element in node_children:
+                if element.shaderName() != "vray:SettingsRenderChannels":
+                    aov = str(element)
+                    render_product = prefix.replace(suffix, aov)
+                    render_element_dict[aov] = render_product
+        return render_element_dict
+
+    def generate_expected_files(self, instance, path):
+        """Create expected files in instance data"""
+
+        dir = os.path.dirname(path)
+        file = os.path.basename(path)
+
+        if "#" in file:
+            pparts = file.split("#")
+            padding = "%0{}d".format(len(pparts) - 1)
+            file = pparts[0] + padding + pparts[-1]
```

Comment:
I understand what it does, but it does seem like an odd way to do this?
I wonder if this is more readable:


```python
import re

def replace(match):
    return "%0{}d".format(len(match.group()))

re.sub("#+", replace, file)
```

Examples:
```python

file = "foo.##.exr"
print(re.sub("#+", replace, file))
# foo.%02d.exr

file = "hello_world_##_####.exr"
print(re.sub("#+", replace, file))
# hello_world_%02d_%04d.exr

file = "bar_#.exr"
print(re.sub("#+", replace, file))
# bar_%01d.exr

file = "bar.exr"
print(re.sub("#+", replace, file))
# bar.exr
```